docs/writing_integration_tests.md

```diff
@@ -0,0 +1,59 @@
+# Writing integration tests
+
+## Developing the test source
+
+If you are trying to write a cachi2 integration test, and wish to run cachi2
+against a local source repo, assuming that
+
+- the image "localhost/cachi2:latest" exists and is a valid cachi2 container
+  image
+- your test source is in "~/temp/cachi2-test", **and it is a valid git repo**
+  (i.e. a valid ".git" directory is present)
+
+executing e.g.
+
+```bash
+podman run --rm -ti -v "~/temp/cachi2-test:~/temp/cachi2-test:z" -w "~/temp/cachi2-test" localhost/cachi2:latest
+```
+
+*should* give you a way to properly process the test source with cachi2
+
+## Running pytest locally
+
+Once you have working test sources, you'll need to commit and push them
+somewhere that pytest can clone them from.
+
+We *strongly* recommend forking this repo and pushing to a new branch named
+after your scenario; this avoids the chicken-and-egg problem of needing a proper
+commit hash in a repo in which you probably don't have merge privileges and
+really need to open a PR first anyway (once your test is complete and passing,
+you can simply open a PR against this repo).
+
+From there, running only *your* test scenario (which will save you *huge*
+amounts of time) looks like e.g.
+
+```bash
+CACHI2_IMAGE=localhost/cachi2:latest pytest -rA -vvvv --confcutdir=tests/integration --log-cli-level=DEBUG tests/integration/test_foo.py::test_foo_package[foo_incorrect_checksum]
+```
+
+This assumes
+
+- you're in the cachi2 repo directory
+- 'pytest' and 'jsonschema' are installed in the cachi2 venv (so that you can
+  run pytest without tox)
+- your test scenario's source is in 'tests/integration/test_foo.py'
+- your test scenario is called `test_foo_package`, and its pytest 'id' is
+  `foo_incorrect_checksum`
```

Comment:
I would expand this some more into individual steps so that the doc feels more like a process wizard rather than a simplified example.
Most importantly I'd like to see an explanantion why tox isn't a good idea for test development --> lack of test output feedback that can be used for test debugging.

Reply:
In progress. I've moved and reworded so we now have a dedicated walkthrough. It's in a separate commit, to be squashed after we hash it out.